Home_Task_9.py

```diff
@@ -0,0 +1,44 @@
+"""
+    Task: Operations with files, REs.
+
+    Description:
+
+write script that will read "input.txt" file
+and find there all characters from english alphabet
+collect these characters and their positions in file
+after info collected this info as text
+write to "output.txt" file in such format
+####################
+<first found letter> -> pos1
+<next_letter> -> pos2
+<next_letter -> pos3
+etc
+#######################
+"""
+
+import re
+
+try:
+#opening both input and output files
+    with open('input.txt', 'r') as fh, open('output.txt', 'w') as fh2:
+        input_data = fh.read()
+
+        # getting the list of letters matching the search criteria
+        letters_found = re.findall('[a-z]', string=input_data)
+
+        # getting matches as iterators
+        iterables_found = re.finditer('[a-z]', string=input_data)
+
+        # retrieving positions of the matches
+        positions_found = [(m.start(0), m.end(0)) for m in iterables_found]
+
+        # matching found letters with the corresponding positions
+        regexp3 = dict(zip(letters_found, positions_found))
```

Comment:
Here and below if you have collected some list with tuples  (m.group(0) m.start(0), m.end(0)) 
Just iterate over it and write to file final result. No sense in dicts and additional zip calls